src/convertToObject.js

```diff
@@ -5,8 +5,21 @@
  *
  * @return {object}
  */
+
 function convertToObject(sourceString) {
-  // write your code here
+  const obj = sourceString
+    .split(';')
+    .filter((line) => line.trim() !== '')
+    .reduce((object, el) => {
+      let [key, value] = el.split(':').map((item) => item.trim());
+
+      key = key.split('').join('').trim();
+      value = value.split('').join('').trim();
+
+      return Object.assign(object, { [key]: value });
```

Comment:
[CODE KNOWLEDGE] - Don't use `Object.assign` inside a `reduce` when you can simply modify and return the accumulator object. This will avoid unnecessary object copies and improve performance.